src/core/infra/repositories/ApiRepository.ts

```diff
@@ -39,15 +39,15 @@ export abstract class ApiRepository {
     operation: string,
     resourceId: number | string = undefined
   ) {
-    let endpoint
-    if (typeof resourceId === 'number') {
-      endpoint = `/${resourceName}/${resourceId}/${operation}`
-    } else if (typeof resourceId === 'string') {
-      endpoint = `/${resourceName}/:persistentId/${operation}?persistentId=${resourceId}`
-    } else {
-      endpoint = `/${resourceName}/${operation}`
+    if (resourceName === 'dataverses') {
+      return `/${resourceName}/${resourceId}`
```

Comment:
I believe ApiRepository should be more generic, this would mean abstracting away any logic specific to particular resource types, thereby avoiding the need for direct checks on these types within the repository. If it's not universally applicable, we might need a refactor. Let's consider opening a new issue for this

What do you think, @GPortas?

Reply:
I agree, the ApiRepository implementation should handle these checks.

Reply:
- #136